components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RequestCountResource.java

```diff
@@ -0,0 +1,142 @@
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.management.apis;
+
+import org.apache.axis2.Constants;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.json.JSONObject;
+import org.wso2.micro.core.util.StringUtils;
+import org.wso2.micro.integrator.initializer.handler.DataHolder;
+import org.wso2.micro.integrator.initializer.handler.transaction.store.TransactionStore;
+import org.wso2.micro.integrator.initializer.handler.transaction.store.connector.RDBMSConnector;
+
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.wso2.micro.integrator.management.apis.Constants.BAD_REQUEST;
+import static org.wso2.micro.integrator.management.apis.Constants.INTERNAL_SERVER_ERROR;
+
+/**
+ * Resource for a retrieving aggregated request count.
+ * <p>
+ * Handles resources in the form "management/reqCount and management/reqCount?year=2020&month=5"
+ */
+public class RequestCountResource extends ApiResource {
+
+    private static Log log = LogFactory.getLog(RequestCountResource.class);
+    private static final String NO_ENTITY_BODY = "NO_ENTITY_BODY";
+
+    /**
+     * Constructor for creating an API Resource.
+     *
+     * @param urlTemplate the url template of the Resource
+     */
+    public RequestCountResource(String urlTemplate) {
+
+        super(urlTemplate);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+
+        Set<String> methods = new HashSet<>();
+        methods.add("GET");
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext synCtx) {
+
+        org.apache.axis2.context.MessageContext axisCtx = ((Axis2MessageContext) synCtx).getAxis2MessageContext();
+        String yearParameter = Utils.getQueryParameter(synCtx, "year");
+        String monthParameter = Utils.getQueryParameter(synCtx, "month");
+
+        if (tryParseInt(yearParameter) != null && tryParseInt(monthParameter) != null) {
+            if (!StringUtils.isEmpty(yearParameter) && !StringUtils.isEmpty(monthParameter)) {
+                return takeRequestCuntOfTheMonth(synCtx, Integer.parseInt(yearParameter),
+                        Integer.parseInt(monthParameter));
+            } else if (StringUtils.isEmpty(yearParameter) && StringUtils.isEmpty(monthParameter)) {
+                Date date = new Date();
+                LocalDate localDate = date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+                return takeRequestCuntOfTheMonth(synCtx, localDate.getYear(), localDate.getMonthValue());
+            }
+        }
+        JSONObject response = Utils.createJsonError("Invalid Input", axisCtx, BAD_REQUEST);
+        axisCtx.removeProperty(org.wso2.micro.integrator.management.apis.Constants.NO_ENTITY_BODY);
```

Comment:
This should have been already included in setJsonPayload method by @sachithKay  .